packages/drawer/src/LionDrawer.js

```diff
@@ -0,0 +1,245 @@
+import { html, uuid } from '@lion/core';
+import { LionCollapsible } from '@lion/collapsible';
+import { drawerStyle } from './drawerStyle.js';
+
+const EVENT = {
+  TRANSITION_END: 'transitionend',
+  TRANSITION_START: 'transitionstart',
+};
+
+export class LionDrawer extends LionCollapsible {
+  static get properties() {
+    return {
+      transitioning: {
+        type: Boolean,
+        reflect: true,
+      },
+      opened: {
+        type: Boolean,
+        reflect: true,
+      },
+      position: {
+        type: String,
+        reflect: true,
+      },
+    };
+  }
+
+  constructor() {
+    super();
+
+    /** @private */
+    this.__toggle = () => {
+      this.opened = !this.opened;
+    };
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (!this.hasAttribute('position')) {
+      this.position = 'left';
+    }
+
+    const uid = uuid();
+
+    if (this._contentNode) {
+      this._contentNode.style.setProperty('display', '');
+    }
+
+    if (this._bottomInvokerNode) {
+      this._bottomInvokerNode.addEventListener('click', this.toggle);
+      this._bottomInvokerNode.setAttribute('aria-expanded', `${this.opened}`);
+      this._bottomInvokerNode.setAttribute('id', `collapsible-invoker-${uid}`);
+
+      if (this._contentNode) {
+        this._bottomInvokerNode.setAttribute('aria-controls', this._contentNode.id);
+      }
+    }
+
+    this.__setBoundaries();
+  }
+
+  /**
+   * Update aria labels on state change.
+   * @param {import('@lion/core').PropertyValues } changedProperties
+   */
+  updated(changedProperties) {
+    if (changedProperties.has('opened')) {
+      this._openedChanged();
+    }
+  }
+
+  static get styles() {
+    return [drawerStyle];
+  }
+
+  __setBoundaries() {
+    const host = this.shadowRoot?.host;
+
+    if (this.position === 'top') {
+      this.minHeight = host ? getComputedStyle(host).getPropertyValue('--min-height') : '0px';
+      this.maxHeight = host ? getComputedStyle(host).getPropertyValue('--max-height') : '0px';
+      this.minWidth = 'auto';
+      this.maxWidth = 'auto';
+    }
+
+    if (this.position === 'left' || this.position === 'right') {
+      this.minWidth = host ? getComputedStyle(host).getPropertyValue('--min-width') : '0px';
+      this.maxWidth = host ? getComputedStyle(host).getPropertyValue('--max-width') : '0px';
+      this.minHeight = 'auto';
+      this.maxHeight = 'auto';
+    }
+
+    setTimeout(() => {
+      const prop = this.position === 'top' ? 'width' : 'height';
+
+      if (this.__contentNode) {
+        this.__contentNode.style.setProperty(prop, '');
+      }
+    });
+  }
+
+  /**
+   * Update aria labels on state change.
+   * @param {String} position
```

Comment:
It's unclear which positions are available.
Also if you change the position, not all styling is set correctly. (tested top -> left).

Reply:
I checked again with designers and we don't need them at the top and bottom simultaneously so I will change this.

Reply:
Why not:

```js
* @param {'top'|'right'|'left'} position
```